src/databricks/labs/ucx/hive_metastore/table_migrate.py

```diff
@@ -93,23 +94,29 @@ def index(self):
     def migrate_tables(self, *, what: What | None = None, acl_strategy: list[AclMigrationWhat] | None = None):
         self._init_seen_tables()
         tables_to_migrate = self._tm.get_tables_to_migrate(self._tc)
-        tasks = []
         if acl_strategy is not None:
             grants_to_migrate = self._gc.snapshot()
             migrated_groups = self._group.snapshot()
             principal_grants = self._principal_grants.get_interactive_cluster_grants()
         else:
             acl_strategy = []
-        for table in tables_to_migrate:
-            grants = []
-            if what is not None and table.src.what != what:
-                continue
-            if AclMigrationWhat.LEGACY_TACL in acl_strategy:
-                grants.extend(self._match_grants(table.src, grants_to_migrate, migrated_groups))
-            if AclMigrationWhat.PRINCIPAL in acl_strategy:
-                grants.extend(self._match_grants(table.src, principal_grants, migrated_groups))
-            tasks.append(partial(self._migrate_table, table.src, table.rule, grants))
-        Threads.strict("migrate tables", tasks)
+        sequencer = TableMigrationSequencer(tables_to_migrate)
+        batches = sequencer.sequence_batches(what)
```

Comment:
What is the benefit of using TableMigrationSequencer here? Most of the logic of TableMigrationSequencer seems to apply to views only, and the `sequencer.sequence_batches` just returns same `tables_to_migrate` when migrating managed and external tables. Would the code be easier to understand and maintain, if we keep the old  `migrate_tables` function and just create a new `migrate_views` function with the TableMigrationSequencer (or call it ViewMigrationSequencer)?

Reply:
I guess we all agree that code duplication is bad.
I actually factorized common code between `migrate_external_tables_sync` and `migrate_dbfs_root_delta_tables` in runtime.py, so I could reuse it for `migrate_views`.
Within `migrate_tables` we have 5-10 lines of code out of 25 that seem overly complex for the table use case. Moving them to a dedicated `migrate_views` method wouldn't remove the complexity, it would just increase code duplication. 
Plus there's something in the way: `what: What | None = None`. Not sure what's the business case for that, but when `what` is None we still need the sequencer (not tested yet). If we can remove the optionality of `what`, then the apparent complexity can be reduced.

Reply:
FWIW `sequencer.sequence_batches` doesn't return the same `tables_to_migrate` when called for the table use case, it applies a filtering.

Reply:
@nfx is there a use case for `what == None` ?

Reply:
Following agreement to remove optionality of `what`, the logic is now separated